bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+
+While channels should not be advertised before they are sufficiently deep, the
+requirement against rebroadcasting only applies if the transaction has not moved
+to a different block.
+
+In order to avoid storing excessively large messages, yet still allow for
+reasonable future expansion, nodes are permitted to restrict rebroadcasting
+(perhaps statistically).
+
+New channel features are possible in the future: backwards compatible (or
+optional) features will have _odd_ feature bits, while incompatible features
+will have _even_ feature bits
+(["It's OK to be odd!"](00-introduction.md#glossary-and-terminology-guide)).
+
+A delay of 12-blocks is used when forgetting a channel on funding output spend
+as to permit a new `channel_announcement_2` to propagate which indicates this
+channel was spliced.
+
+## The `node_announcement_2` Message
+
+This gossip message, like the legacy `node_announcement` message, allows a node 
+to indicate extra data associated with it, in addition to its public key. 
+To avoid trivial denial of service attacks, nodes not associated with an already
+known channel (legacy or taproot) are ignored.
+
+Unlike the legacy `node_announcement` message, this message makes use of a 
+Schnorr signature instead of an ECDSA one. This will allow nodes to be backed 
+by multiple keys since MuSig2 can be used to construct the single signature.
+
+The other two main differences from the legacy message are that the timestamp 
+is now a block height and that the message is purely TLV based.
+
+1. type: xxx (`node_announcement_2`)
+2. data: (tlv_stream):
+    1. type: 0 (`announcement_sig`)
+    2. data:
+        * [`bip340_sig`:`bip340_sig`]
+    3. type: 1 (`features`)
+    4. data:
+        * [`...*byte`: `features`]
+    5. type: 2 (`timestamp`)
+    6. data:
+        * [`u32`: `block_height`]
+    7. type: 3 (`node_ID`)
+    8. data: 
+        * [`point`:`node_id`]
+    9. type: 4 (`color`)
+    10. data:
+         * [`rgb_color`:`rgb_color`]
+    11. type: 5 (`alias`)
+    12. data:
+         * [`...*utf8`:`alias`]
+    13. type: 6 (`ipv4_addrs`)
+    14. data: 
+         * [`...*ipv4_addr`: `ipv4_addresses`]  
+    15. type: 7 (`ipv6_addrs`)
+    16. data:
+         * [`...*ipv6_addr`: `ipv6_addresses`]
+    17. type: 8 (`tor_v3_addrs`)
+    18. data: 
+         * [`...*tor_v3_addr`: `tor_v3_addresses`]
+    19. type: 9 (`dns_hostname`)
+    20. data:
+         * [`dns_hostname`: `dns_hostname`]
+
+The following subtypes are defined:
+
+1. subtype: `rgb_color`
+2. data:
+   * [`byte`:`red`]
+   * [`byte`:`green`]
+   * [`byte`:`blue`]
+
+3. subtype: `ipv4_addr`
+4. data:
+    * [`u32`:`addr`]
+    * [`tu16`:`port`]
+
+5. subtype: `ipv6_addr`
+6. data:
+    * [`16*byte`:`addr`]
+    * [`tu16`:`port`]
+ 
+7. subtype: `tor_v3_addr`
+8. data:
+    * [`35*utf8`:`onion_addr`]
+    * [`tu16`:`port`]
+
+9. subtype: `dns_hostname`
+10. data:
+     * [`...*utf8`:`hostname`]
+     * [`tu16`:`port`]
```

Comment:
[From Rusty](https://github.com/ellemouton/bolts/pull/2#discussion_r1134932702):

And if port is 0, say it's 9735?

Reply:
same here as for the above re defaults